src/BenchmarkDotNet/Toolchains/IToolchain.cs

```diff
@@ -14,5 +17,6 @@ public interface IToolchain
         bool IsInProcess { get; }
 
         bool IsSupported(BenchmarkCase benchmarkCase, ILogger logger, IResolver resolver);
+        IEnumerable<ValidationError> CheckSupport(BenchmarkCase benchmarkCase, ILogger logger, IResolver resolver);
```

Comment:
since this method returns `IEnumerable<ValidationError>`, it  could be called `Validate`. Moreover since we should change its responsibility to just return the errors (not report them), and don't pass `ILogger` instance to it:


```suggestion
        IEnumerable<ValidationError> Validate(BenchmarkCase benchmarkCase, IResolver resolver);
```

Reply:
Actually the responsibility remains the same, but it returns an IEnumerable<ValidationError> instead of just a bool, so that the client knows specifically what happened.
Removing the logger means that we would have to create some generic approach for "printing" toolchain information, as the logger is used inside the toolchain implementations, in places like this https://github.com/dotnet/BenchmarkDotNet/blob/master/src/BenchmarkDotNet/Toolchains/InProcess/InProcessValidator.cs#L107.
Renamed to Validate